linalg/solvers.cpp

```diff
@@ -1641,6 +1641,138 @@ void NewtonSolver::Mult(const Vector &b, Vector &x) const
    final_norm = norm;
 }
 
+void LBFGSSolver::Mult(const Vector &b, Vector &x) const
```

Comment:
This is great! I will definitely use this.

Reply:
Thanks @jamiebramwell  
p.s. I remember you suggested L-BFGS when we were looking at gradient based solvers back when I was a summer student in 2018. Thanks for the suggestion!